__test__/main.test.ts

```diff
@@ -6,26 +6,97 @@ import * as osx from "../src/setup-ros-osx";
 import * as windows from "../src/setup-ros-windows";
 import * as utils from "../src/utils";
 
-utils.lib.exec = jest.fn(async (commandLine: string,
-    args?: string[],
-    options?: im.ExecOptions,
-    log_message?: string) => 
-    {
-        const argsAsString = (args || []).join(" ");
-        const message = log_message || `Invoking "${commandLine} ${argsAsString}"`;
-        return core.group(message, async () => {
-            return 0;
-        });
-    });
-
-test('run Windows workflow', async () => {
-    return windows.runWindows();
-})
 
-test('run Linux workflow', async () => {
-    return linux.runLinux();
-})
+describe('basic workflow tests', () => {
+    let linesOfExec: string[] = [];
+    beforeAll(() => {
+        jest
+            .spyOn(utils, 'exec')
+            .mockImplementation(async (
+                commandLine: string,
+                args?: string[],
+                options?: im.ExecOptions,
+                log_message?: string) => {
+                    let flatArgs = ''
+                    if (args) {
+                        flatArgs = args.map(i => i).join(' ')
+                    }
+                    linesOfExec.push(`${commandLine} ${flatArgs}`)
+                    return 0;
+                });
+    })
+
+    beforeEach(() => {
+        linesOfExec = [];
+    })
+
+    afterAll(() => {
+        jest.restoreAllMocks();
+    })
 
-test('run macOS workflow', async () => {
-    return osx.runOsX();
+    it('run Linux workflow', async () => {
+        await linux.runLinux();
+        console.log(linesOfExec);

```

Comment:
Any reason to remove the logging from the mock implementation?

Reply:
No any particular reasons for the change. Now I mock the `action_exec.exec` and we can just reuse the logging from the `utils.exec`.